packages/sanity/src/core/studio/StudioTelemetryProvider.tsx

```diff
@@ -0,0 +1,56 @@
+import {createBatchedStore, createSessionId} from '@sanity/telemetry'
+import {TelemetryProvider} from '@sanity/telemetry/react'
+import React, {ReactNode, useEffect, useMemo} from 'react'
+import arrify from 'arrify'
+import {SANITY_VERSION} from '../version'
+import {Config} from '../config'
+import {useClient} from '../hooks'
+import {StudioMount} from './__telemetry__/studio.telemetry'
+
+const sessionId = createSessionId()
+
+// Wrap the app in a TelemetryProvider
+// This will enable usage of the `useTelemetry()` hook
+export function StudioTelemetryProvider(props: {children: ReactNode; config: Config}) {
+  const client = useClient({apiVersion: 'v2023-12-18'})
+
+  const projectId = client.config().projectId
+  const store = useMemo(() => {
+    return createBatchedStore(sessionId, {
+      // submit any pending events every <n> ms
+      flushInterval: 30000,
+
+      // implements user consent resolving
+      resolveConsent: () => client.request({uri: '/intake/telemetry-status'}),
+
+      // implements sending events to backend
+      sendEvents: (batch) =>
+        client.request({
+          uri: '/intake/batch',
+          method: 'POST',
+          json: true,
+          body: {projectId, batch},
+        }),
+      // opts into a different strategy for sending events when the browser close, reload or navigate away from the current page
+      sendBeacon: (events) =>
+        navigator.sendBeacon(
+          client.getUrl('/intake/batch'),
+          new Blob([JSON.stringify(events)], {type: 'text/plain'}),
```

Comment:
If `sendBeacon` is locked to text/plain, we'll need to handle this new content-type

Reply:
Service was updated

Reply:
It's not strictly limited, but using anything else will trigger a CORS preflight, which we probably want to avoid spending time on in this case.